src/domains/setting/pages/Settings/Settings.tsx

```diff
@@ -1,96 +1,24 @@
 import { Page, Section } from "app/components/Layout";
 import { SideBar } from "app/components/SideBar";
-import { useActiveProfile, useReloadPath } from "app/hooks";
-import { toasts } from "app/services";
-import React, { useState } from "react";
-import { useForm } from "react-hook-form";
-import { useTranslation } from "react-i18next";
-import { matchPath, Prompt } from "react-router-dom";
-
-import { availableSettings } from "./available-settings";
+import { useActiveProfile } from "app/hooks";
+import { useSettingsMenu } from "domains/setting/hooks/use-settings-menu";
+import { Export, General, PasswordSettings } from "domains/setting/pages/Settings";
+import React from "react";
 
 export const Settings = () => {
-	const [activeSettings, setActiveSettings] = useState("General");
-
-	const reloadPath = useReloadPath();
-
-	const form = useForm({ mode: "onChange" });
-	const { register, errors, formState, reset } = form;
-	const { isDirty } = formState;
-
 	const activeProfile = useActiveProfile();
-
-	const { t } = useTranslation();
-
-	const settingsItems = [
-		{
-			itemKey: "General",
-			label: t("SETTINGS.GENERAL.MENU_ITEM"),
-			icon: "SettingsHorizontal",
-		},
-		{
-			itemKey: "Password",
-			label: t("SETTINGS.PASSWORD.MENU_ITEM"),
-			icon: "Lock",
-		},
-		{
-			itemKey: "Export",
-			label: t("SETTINGS.EXPORT.MENU_ITEM"),
-			icon: "Export",
-		},
-	];
-
-	const handleSuccess = (message?: string) => {
-		reloadPath();
-		reset();
-
-		toasts.success(message || t("SETTINGS.GENERAL.SUCCESS"));
-	};
-
-	const handleError = (errorMessage: string, message?: string) => {
-		toasts.error(`${message || t("COMMON.ERROR")}: ${errorMessage}`);
-	};
-
-	const getPromptMessage = (location: any) => {
-		/* istanbul ignore next */
-		const pathname = location.pathname || location.location?.pathname;
-
-		const matchCurrent = matchPath(pathname, {
-			path: "/profiles/:profileId/settings",
-		});
-
-		const isReload = matchCurrent !== null;
-
-		if (isReload) {
-			return true;
-		}
-
-		if (isDirty) {
-			return "block";
-		}
-
-		return true;
-	};
-
-	const renderSettings = () => {
-		const ActiveSettings = availableSettings[activeSettings];
-
-		return (
-			<ActiveSettings
-				formConfig={{ context: form, register, errors }}
-				onSuccess={handleSuccess}
-				onError={handleError}
-			/>
-		);
-	};
```

Comment:
I think we can still use this function to render setting page dynamically
like this (tested)
```
const renderSettings = () => {
	const ActiveSettings = availableSettings[activeSettings];

	return (
		<ActiveSettings/>
	);
};
```

Reply:
@w3ea I removed the renderSettings for readability as it seemed more clear to know which component is rendering :hmhm: . The Settings component now renders the 3 tabs as: 
```jsx
export const Settings = () => {
	const activeProfile = useActiveProfile();
	const { menuItems, activeSettings, setActiveSettings } = useSettingsMenu();

	return (
		<Page
			profile={activeProfile}
			sidebar={<SideBar items={menuItems} activeItem={activeSettings} handleActiveItem={setActiveSettings} />}
		>
			<Section innerClassName="px-12">
				{activeSettings === "General" && <General />}
				{activeSettings === "Password" && <PasswordSettings />}
				{activeSettings === "Export" && <Export />}
			</Section>
		</Page>
	);
};

```

Reply:
yeah, i got that
but for adding a new page we need to change two place
[here ](https://github.com/ArkEcosystem/desktop-wallet/pull/3969/files#diff-fd2e0e758c3d350d8b2e7baa8f38e25cc6bc7f18d92151df18ba3942627cc29cR8-R24) and [here](https://github.com/ArkEcosystem/desktop-wallet/pull/3969/files#diff-464370bbc0ddbe87686580ee018bc24e908d5083e34ac34218f9aa1524992a7bR18-R20)
it's totally ok, it was just a suggestion